### PR TITLE
Make `build-graph` recur until all transitive deps are analyzed

### LINKED ISSUE #375: Latest clerk errors on notebook analyses, ->hash must be ifn?
On https://github.com/mentat-collective/mathbox.cljs, branch `sritchie/new_release`, when I build my static bundle with `bb publish-local` I see the following output:

```
🔬 Analyzing… ->hash must be `ifn?` {:->hash nil, :codeblock {:form (show-sci [mathbox/Mathbox {:style {:height 500px, :width 100%}, :init {:background-color 16777215, :camera-position [3 2 5], :scale 500}} [mb/Cartesian {} [mb/Transform4 {:matrix (let [a (/ (* (:projectionAngle (clojure.core/deref !state)) 2 Math/PI) 360.0)] [1 0 0 (Math/cos a) 0 1 0 0 0 0 1 (Math/sin a) 0 0 0 1])} [mb/Area {:rangeX [-3 1], :rangeY [(* -2 Math/PI) (* 2 Math/PI)], :width 129, :height 65, :channels 4, :expr (fn [emit x y _i _j] (let [r (Math/exp x)] (emit (* r (Math/cos y)) (* r (Math/sin y)) x y)))}] [mb/Surface {:color (:surface (clojure.core/deref !state)), :zBias -0.25}] [mb/Line {:color (:line (clojure.core/deref !state)), :width 4}] [mb/Transpose {:order yxzw}] [mb/Line {:color (:line (clojure.core/deref !state)), :width 4}]]]]), :ns-effect? false, :freezable? true, :no-cache? false, :deps #{nextjournal.clerk/with-viewer mentat.clerk-utils.show/show-sci nextjournal.clerk/mark-presented}, :file /Users/sritchie/.gitlibs/libs/io.github.mentat-collective/clerk-utils/e40cdf79edf45748f5a2c6752c8156b3d240b087/src/mentat/clerk_utils/show.cljc, :id mathbox.examples.math.exponential/anon-expr-5drxcaGZr7zvK46gPq5a5FKTUYBvdG}}
Done in 600.672ms. ✅
```

This also comes up at the REPL when I save the file containing this form, but only sometimes.

I'm not sure how `:->hash nil` ends up in there. This is happening on version 5cf3982b29c25a6fe70ee4d4444293c5320ff697, and was NOT happening a week ago with the same code, though I haven't done a proper bisection to figure out when this came in.

### PR DESCRIPTION
**Pull request by philomates**
Until now Clerk did not analyze the full transitive dependency graph which could lead to Clerk not detecting a change properly. Analysis is now recursive which means it's taking a bit longer initially. We cache analysis results per file in memory so subsequent analysis should be fast. We will follow up with visualizing the progress of analysis & execution.

Also discovered cases where classes instead of symbols could end up in the dependency graph and introduced normalization to symbols.

This also gets rid of the `->hash must be ifn?` warning which fixes #375.

### CODE REVIEW

File: src/nextjournal/clerk/analyzer.clj [hunk 1]
```diff
@@ -419,15 +461,54 @@
...
+
+         (-> (cond-> doc
+               (not (:graph doc)) analyze-doc)
+             (assoc :analyzed-file-set (cond-> #{} (:file doc) (conj (:file doc))))
+             (assoc :counter 0))]
+    (let [unhashed (unhashed-deps ->analysis-info)
+          loc->syms (apply dissoc
+                           (group-by find-location unhashed)
+                           analyzed-file-set)]
+      (prn :build-graph counter :analyzed-file-set analyzed-file-set)
```
Comment (borkdude) [inline src/nextjournal/clerk/analyzer.clj:474]: There is still a prn here, is that intended?

File: src/nextjournal/clerk/analyzer.clj [hunk 2]
```diff
@@ -419,15 +461,54 @@
...
+      (if (and (seq loc->syms) (< counter 10))
+        (recur (-> (reduce (fn [g [source symbols]]
+                             (prn :source source :symbols symbols :ana (when (or (nil? source)
```
Comment (borkdude) [inline src/nextjournal/clerk/analyzer.clj:477]: Also one here

File: src/nextjournal/clerk/analyzer.clj (hunk 1 above)
Comment (mk) [reply src/nextjournal/clerk/analyzer.clj:474]: added those to debug windows on CI. Will remove them before merging.

**Comment by borkdude** [comment]
@mk I pushed a fix for Windows. I also added an alias for the cognitect test runner which I find easier to work with, I hope that's ok.

**Pull request merged**